### PR TITLE
Add script to compare benchmark results

### PR DESCRIPTION
**Pull request by shabalind**
This is same pull request as #76, only against `main` branch.

### CODE REVIEW

**Comment by dabrahams** [comment]
Glad to see this proceeding!

File: Scripts/compare.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+    for i, benchmark in enumerate(parsed["benchmarks"]):
+        require("name" in benchmark,
+                "{}: benchmark #{}: missing key 'name'.".format(file_name, i))
+
+        for k, v in benchmark.items():
+            if k != "name":
+                is_num = isinstance(v, int) or isinstance(v, float)
+                template = "{}: benchmark #{}: values must be numbers."
+                require(is_num, template.format(file_name, i))
```
Comment (compnerd) [inline Scripts/compare.py]:
```suggestion
            if k == "name": continue
            is_num = isinstance(v, int) or isinstance(v, float)
            template = "{}: benchmark #{}: values must be numbers."
            require(is_num, template.format(file_name, i))
```

File: Scripts/compare.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    return (confs, values)
+
+
+def geomean(values):
+    """Compute geometric mean for the given sequence of values."""
+
+    product = 1.0
+    for value in values:
+        product *= value
+    return product**(1.0/len(values))
```
Comment (compnerd) [inline Scripts/compare.py]:
```suggestion
    return product**(1.0 / len(values))
```

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
@googlebot I consent.

**Pull request merged**